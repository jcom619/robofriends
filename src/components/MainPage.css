body {
  margin: 0;
  background: rgb(2,0,36);
  background: linear-gradient(90deg, rgba(2,0,36,1) 0%, rgba(14,90,104,1) 0%, rgba(0,212,255,1) 100%);
}

h1 {
  font-family: phosphate;
  font-weight: 200;
  font-size: 50px;
  color: lightblue;
  text-shadow: 0px 0px 15px black;
}
