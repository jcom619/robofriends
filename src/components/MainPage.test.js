import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<MainPage { ...mockProps } />)
})

it('renders MainPage without crashing', () => {
  expect(wrapper.debug()).toMatchSnapshot();
})

it('filters robots correctly', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'john',
      email: 'john@email.com'
    }],
    searchField: 'john',
    isPending: false
  }
  const wrapper2 = shallow(<MainPage { ...mockProps2 } />)
  expect(wrapper2.instance().filterRobots()).toEqual([{
    id: 3,
    name: 'john',
    email: 'john@email.com'
  }]);
})


it('filters robots correctly 2', () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'cindy',
      email: 'cindy@email.com'
    }],
    searchField: 'a',
    isPending: false
  }
  const filteredRobots = []
  const wrapper3 = shallow(<MainPage { ...mockProps3 } />)
  expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
})
