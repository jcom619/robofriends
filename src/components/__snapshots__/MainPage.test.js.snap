// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
"<div className=\\"tc f1\\">
  <Header />
  <SearchBox searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundry>
      <CardList robots={{...}} />
    </ErrorBoundry>
  </Scroll>
</div>"
`;
